embedded-kafka/src/main/scala/net/manub/embeddedkafka/Consumers.scala

```diff
@@ -0,0 +1,56 @@
+package net.manub.embeddedkafka
+
+import java.util.Properties
+
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.serialization.Deserializer
+
+/** Utility trait for easily creating Kafka consumers and accessing their consumed messages. */
+trait Consumers {
+  /** Loaner pattern that allows running a code block with a newly created consumer.
+    * The consumer's lifecycle will be automatically handled and closed at the end of the
+    * given code block.
+    *
+    * @param block the code block to be executed with the instantiated consumer
+    *              passed as an argument
+    * @tparam K the type of the consumer's Key
+    * @tparam V the type of the consumer's Value
+    * @tparam T the type of the block's returning result
+    * @return the result of the executed block
+    */
+  def withConsumer[K: Deserializer, V: Deserializer, T](block: KafkaConsumer[K, V] => T)
+                                                       (implicit config: EmbeddedKafkaConfig): T = {
+    val consumer = newConsumer[K, V]()
+    val result = block(consumer)
```

Comment:
can `block()` ever throw an exception? as probably we don't know in advance it would be better to wrap `block(consumer)` into a `try / finally` block in order to close the consumer regardless of what happens.